File: BullyAlgorithm/src/main/java/src/core/BullyAlgorithm.java
package src.core;

public class BullyAlgorithm {
    private Routines routines;
    private ProcessContainer processContainer;

    public static void main(String[] args) {
        new BullyAlgorithm();
    }

    public BullyAlgorithm() {
        routines = new Routines();
        processContainer = new ProcessContainer();

        routines.startNewRoutine(2, processContainer::requestToCoordinator);
        routines.startNewRoutine(3, processContainer::createProcess, true);
        routines.startNewRoutine(8, processContainer::killRandomProcess);
        routines.startNewRoutine(10, processContainer::killCoordinator);
    }


}

File: BullyAlgorithm/src/main/kotlin/core/BullyAlgorithm.kt
import actions.KillCoordinator
import actions.KillRandomProcess
import actions.NewProcess
import actions.RequestCoordinator

class BullyAlgorithm {

    var processList: LinkedHashMap<Long, Process> = linkedMapOf()

    private val newProcess: NewProcess = NewProcess(30)
    private val requestCoordinator: RequestCoordinator = RequestCoordinator(25)
    private val killCoordinator: KillCoordinator = KillCoordinator(100)
    private val killRandomProcess: KillRandomProcess = KillRandomProcess(80)

    init {
        killCoordinator.start()
        killRandomProcess.start()
        newProcess.start()
        requestCoordinator.start()
    }

}